### DISCUSSION TITLE
Adding gatsby-source-s3 to the Gatsby monorepo

### DISCUSSION THREAD

**Discussion by robinmetral** [gatsbyjs/gatsby#34050] [Ideas / Feature Requests] [answered]
Hi people—

For the past two years, I've been maintaining a source plugin to pull images from AWS S3 buckets. It enables Gatsby to process images with sharp, while hosting images off of GitHub—useful for image-heavy sites and blogs that don't use a CMS.

## Context

I didn't create the first ever `gatsby-source-s3`: [the first one](https://github.com/DSchau/gatsby-source-s3/) was made by DSchau, but it stopped being maintained long ago (last commit was in late 2019) and I created mine to fix existing issues with it. There's also jessestuart's [`gatsby-source-s3-image`](https://github.com/jessestuart/gatsby-source-s3-image), which was written in TypeScript but also stopped being maintained a few years ago (last commit excluding automatic dep updates was in late 2019).

So my version of `gatsby-source-s3` took aspects from these two unmaintained plugins and improved upon them.

Some key aspects:

- It supports pulling from multiple and/or private S3 buckets
- It's fully compatible with Gatsby v4 (recently updated)
- It has a nice e2e test suite (bootstraps a real Gatsby site pulling from real S3 buckets inside GitHub actions)
- TypeScript

👉 Find it on GitHub at [robinmetral/gatsby-source-s3](https://github.com/robinmetral/gatsby-source-s3)

## Proposal

Now for the catch: I'm not using Gatsby very much any more, and I don't think I can keep up with future maintenance and potential updates for the plugin.

Which is why I'm wondering if we could consider **moving the source plugin into the Gatsby monorepo**, alongside other source packages.

This would have several benefits:
- ensure maintenance of the plugin in the future along with the framework
- encourage community contributions to the plugin (there were 3 contributors to the plugin so far)
- improve visibility for the plugin: people continue to install the other, unmaintained `gatsby-source-s3` plugins, only to realize that they have bugs or doesn't work with their Gatsby version

Any thoughts here?

**Comment by LekoArts** [comment, marked as answer]
Hi!

We won't be adding more packages to our monorepo (we rather try to minimize the count) but we'll be working on properly highlighting, supporting and incorporating https://github.com/gatsby-uc/plugins into the community & governance. tl;dr this is the place where I'd propose you put your plugin :)

**Reply by robinmetral** [reply]
Oh nice, this looks exactly like the home I was looking for for this plugin 🎉 Thanks for sharing! I'll bring the discussion over to that repo
